gateway/api/serializers.py

```diff
@@ -188,3 +188,67 @@ class RuntimeJobSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = RuntimeJob
+
+
+class RunProgramSerializer(serializers.Serializer):
+    """
+    Program serializer for the /run end-point
+    """
+
+    title = serializers.CharField(max_length=255, allow_blank=False)
+    entrypoint = serializers.CharField(max_length=255)
+    artifact = serializers.FileField(allow_empty_file=False, use_url=False)
+    dependencies = serializers.CharField(allow_blank=False)
+    arguments = serializers.CharField(allow_blank=False)
+    config = serializers.CharField(allow_blank=False)
+
+    def to_representation(self, instance):
+        """
+        Transforms string `config` to JSON
+        """
+        representation = super().to_representation(instance)
+        representation["config"] = json.loads(representation["config"])
+        return representation
+
+    def retrieve_one_by_title(self, title, author):
+        """
+        This method returns a Program entry if it finds an entry searching by the title, if not None
+        """
+        return (
+            Program.objects.filter(title=title, author=author)
+            .order_by("-created")
+            .first()
+        )
+
+    def create(self, validated_data):
+        pass
+
+    def update(self, instance, validated_data):
+        pass
+
+
+class RunProgramModelSerializer(serializers.ModelSerializer):
```

Comment:
These 2 serializers are confusing for me.  Is it possible merge there 2 into 1?

Reply:
Hahaha I just had a meeting with @IceKhan13 due to this. I totally agree with you. They cannot be merged though, because it seems you cannot configure a field in a `ModelSerializer` (or a `Serializer` that creates an instance) that it's not in the model. That's why I needed to create two models:
- One serializer to check the request
- One model serializer to safe the `instance` without taking into account additional fields (read_only=True)

Something that I want to take a look more in-depth as a next step is to create a better structure like:
- serializers
  - run: for the end-point /run
    - BodySerializers: these are serializers that check the request.data
    - ModelSerializers: these are the serializers that receives the data from the BodySerializers and create the instances
- views
  - run: it uses its own BodySerializers and ModelSerializers

I think it integrates well with swagger documentation. But there were too many changes for this set of pull requests and I'm just trying to understand how Django works.

Reply:
Btw @akihikokuroda , what we can do it is rename them to something that it has more sense. Any proposal is welcome.

Reply:
I can not come up with any good names.